### PR TITLE
Remove ID attribute from JSON data <script> tag

### LINKED ISSUE #30: id attribute on script tag breaks iXBRL validity
The "taxonomy-data" block currently includes an `id` attribute on the `<script>` tag.  This is not valid according to the iXBRL schema.

We should fix this, although removing it will require a different approach to locating the element from the JavaScript.

### PR DESCRIPTION
**Pull request by paulwarren-wk**
ID attribute on script tag is invalid according to the iXBRL/XHTML schema.  We now locate the taxonomy data by its position in the document, and report an error in the loading mask if the element is not what we expect.

Fixes #30

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(2) Security relevant changes were detected</summary>
<li>Watched keyword <code>etree</code> in <code>iXBRLViewerPlugin/iXBRLViewer.py</code> line(s) ['262'] added</li>
<li>Watched keyword <code>etree</code> in <code>iXBRLViewerPlugin/iXBRLViewer.py</code> line(s) ['262'] removed</li>
</details>

## Action Items
* Request infosec review via Slack: #support-infosec
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: iXBRLViewerPlugin/viewer/src/js/index.js [hunk 1]
```diff
@@ -41,6 +41,14 @@ function reparentDocument() {
 
 }
 
+function getTaxonomyData() {
+    var elt = document.body.lastElementChild;
+    if (elt.tagName.toUpperCase() != 'SCRIPT' || elt.getAttribute("type") != 'application/json') {
```
[the remark this answers is not in the document]
Comment (paulwarren-wk) [reply iXBRLViewerPlugin/viewer/src/js/index.js]: XHTML and XML validity is required as for documents that rely on being rendered in XHTML mode, we need browsers to be able to treat them as such.  We have been targeting iXBRL validity too, as it's a nice feature that the viewer files can be opened directly in iXBRL tools.

I'll look at other ways to do this, as I agree that this approach is unnecessarily fragile.  I think what you propose should work.  Alternatively, we could have it search for elements of the right name and type, parse them as JSON, and then look for a particular property in the resulting JSON object.

[the remark this answers is not in the document]
Comment (brettkail-wk) [reply iXBRLViewerPlugin/viewer/src/js/index.js]: I believe browsers only require XML well-formedness (not XML validity).  I agree it's a nice feature if the viewer file can be opened directly in iXBRL tools, but it's a trade-off.
> Alternatively, we could have it search for elements of the right name and type, parse them as JSON, and then look for a particular property in the resulting JSON object

Yes, that would be an option.  Or perhaps another option is to use a custom `application/vnd.` media type and then load data from the script of that type?

**Comment by brettkail-wk** [comment]
@paulwarren-wk Merge conflict.

File: iXBRLViewerPlugin/viewer/src/js/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulwarren-wk) [reply iXBRLViewerPlugin/viewer/src/js/index.js]: This is done in e84cabe.  It is uses "application/x.ixbrl-viewer+json" which I think is more technically correct for a custom but unregistered mime type.

**Comment by brettweiss-wf** [comment]
Security +1

**Comment by derekgengenbacher-wf** [comment]
QA +10 Unit tests pass successfully. Builds successfully. Local smoke test passed successfully.

@Workiva/release-management-p

**Pull request merged**